spec/asana_add_comment_spec.rb

```diff
@@ -0,0 +1,98 @@
+require "climate_control"
+
+describe Fastlane::Actions::AsanaAddCommentAction do
+  describe "#run" do
+    before do
+      @asana_client_stories = double
+      asana_client = double("Asana::Client")
+      allow(Asana::Client).to receive(:new).and_return(asana_client)
+      allow(asana_client).to receive(:stories).and_return(@asana_client_stories)
+    end
+
+    it "does not call task id extraction if task id provided" do
+      allow(@asana_client_stories).to receive(:create_story_for_task).and_return(double)
+      expect(Fastlane::Actions::AsanaExtractTaskIdAction).not_to receive(:run)
+      test_action(task_id: "123", comment: "", workflow_url: "")
+    end
+
+    it "extracts task id if task id not provided" do
+      allow(@asana_client_stories).to receive(:create_story_for_task).and_return(double)
+      expect(Fastlane::Actions::AsanaExtractTaskIdAction).to receive(:run)
+        .and_return("9999")
+      test_action(task_url: "https://app.asana.com/0/753241/9999", comment: "", workflow_url: "")
+    end
+
+    it "shows error if both task id and task url are not provided" do
+      expect(Fastlane::UI).to receive(:user_error!).with("Both task_id and task_url cannot be nil. At least one must be provided.")
+      test_action
+    end
+
+    it "shows error if both comment and template_name are not provided" do
+      expect(Fastlane::UI).to receive(:user_error!).with("Both comment and template_name cannot be nil. At least one must be provided.")
+      test_action(task_id: "123")
+    end
+
+    it "shows error if comment is provided but workflow_url is not" do
+      expect(Fastlane::UI).to receive(:user_error!).with("If comment is provided, workflow_url cannot be nil")
+      test_action(task_id: "123", comment: "")
+    end
+
+    it "shows error if provided template does not exist" do
+      allow(File).to receive(:read).and_raise(Errno::ENOENT)
+      expect(Fastlane::UI).to receive(:user_error!).with("Error: The file 'non-existing.yml' does not exist.")
+      expect(@asana_client_stories).not_to receive(:create_story_for_task)
+      test_action(task_id: "123", template_name: "non-existing")
+    end
+
+    it "correctly substitutes all variables" do
+      template_content = "<h2>${ASSIGNEE_ID} is publishing ${TAG} hotfix release</h2>"
+      ClimateControl.modify(
+        ASSIGNEE_ID: '12345',
+        TAG: 'v1.0.0'
+      ) do
+        result = Fastlane::Actions::AsanaAddCommentAction.process_template_content(template_content)
+        expected_output = "<h2>12345 is publishing v1.0.0 hotfix release</h2>"
+        expect(result).to eq(expected_output)
+      end
+    end
+
+    it "removes newlines and leading/trailing spaces" do
```

Comment:
Could you add one more test similar to this one? It's about removing only 1 consecutive newline, i.e.
```
Hello\n  \n  World
```

should render 
```
Hello\nWorld
```

(or something similar 🤪 but we have a bunch of templates that add 2 newlines to end up with 1 newline in Asana)